Subject: dont force no alignment for s390
From: Michael Tokarev <mjt@tls.msk.ru>
Bugs-Debian: http://bugs.debian.org/681760
Forwarded: no

This fix hasn't been forwarded upstream, because might be more
correct fix is to use the ALIGN* marks properly in the declarations
too, or just remove whol ALIGN* mess:

 http://lists.busybox.net/pipermail/busybox/2012-July/078155.html
 http://lists.busybox.net/pipermail/busybox/2012-July/078163.html

--- a/include/platform.h
+++ b/include/platform.h
@@ -284,7 +284,13 @@
 #define fdprintf dprintf
 
 /* Useful for defeating gcc's alignment of "char message[]"-like data */
-#if 1 /* if needed: !defined(arch1) && !defined(arch2) */
+/* #if 1 /o if needed: !defined(arch1) && !defined(arch2) */
+/* on s390 and s390x, GCC may generate incorrect code when accessing
+ * variables defined with ALIGN* but declared without -- assuming alignment
+ * on access which isn't actually present.  The right fix is to use ALIGN*
+ * properly in all declarations too, but there are quite some of such places.
+ */
+#if !defined(__s390__)
 # define ALIGN1 __attribute__((aligned(1)))
 # define ALIGN2 __attribute__((aligned(2)))
 # define ALIGN4 __attribute__((aligned(4)))
